### PR TITLE
Mask sensitive props for task/job

### LINKED ISSUE #2469: Add TaskDefinition support to `ArgumentSanitizer`
On the task side the sanitization support is mostly not there, only `TaskExecutions` are sanitized in the `TaskExecutionController`.

We also need:

- Add support to sanitize Task Arguments to `ArgumentSanitizer`
- Add a TaskDefinition-specific `dslConverter`, similar to `StreamDefinitionToDslConverter`
- Add unit tests

Ensure sanitization works for composed tasks.

See also gh-2461

### LINKED ISSUE #2516: Composed-Tasks: Security arguments for tasks are printed in cleartext
Description:
Spring cloud data flow users could see values of security arguments --dataflow-server-username and  --dataflow-server-password through WebUI or in Log of SCDF.

Release versions:
I have found this problem in 1.7.0, 1.7.0RC1 and earlier.

Steps to reproduce:
-Enable LDAP
-Create any Task
-Start composed task with arguments --dataflow-server-username and  --dataflow-server-password.
-Look at Job tab in WebUI or SCDF logs.

Screenshots:
image
image

But you couldn't see password in Tasks tab:
image

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Move ArgumentSanitizer to REST resource module
  - Add masking for JobParameters
    - for JobExecutionResource
  - Add TaskDefinitionToDSL converter
  - Add masking of task definition
  - Update Job commands results

Resolves #2469
Resolves #2516

### CODE REVIEW

File: spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/TaskDefinitionToDslConverter.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.Map;
+
+import org.apache.commons.lang.StringEscapeUtils;
+
+/**
+ * Reverse engineers a {@link TaskDefinition} into a semantically equivalent DSL text representation.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public class TaskDefinitionToDslConverter {
```
Comment (tzolov) [inline spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/TaskDefinitionToDslConverter.java]: Given how tricky the reverse engineering could be we must have a proper Test class for this converter. 
Similar to StreamDefinitionToDslConverterTests we should validate the behavior with different expressions, characters or regexes.

File: spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/support/ArgumentSanitizer.java [hunk 2]
```diff
@@ -110,6 +116,25 @@ public String sanitize(String key, String value) {
 		return value;
 	}
 
+	/**
+	 * Replaces the sensitive String values in the JobParameter value.
+	 *
+	 * @param jobParameters the original job parameters
+	 * @return the sanitized job parameters
+	 */
+	public JobParameters sanitizeJobParameters(JobParameters jobParameters) {
```
Comment (tzolov) [inline spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/support/ArgumentSanitizer.java]: Add ArgumentSanitizerTest test for `sanitizeJobParameters`

File: spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/support/ArgumentSanitizer.java [hunk 3]
```diff
@@ -134,10 +159,10 @@ public String sanitizeStream(StreamDefinition streamDefinition) {
 	 * @return Task definition text that has sensitive data redacted.
 	 */
 	public String sanitizeTask(TaskDefinition taskDefinition) {
-		final TaskDefinition sanitizedTaskDefinition = TaskDefinition.TaskDefinitionBuilder.from(taskDefinition)
+		TaskDefinition sanitizedTaskDefinition = TaskDefinition.TaskDefinitionBuilder.from(taskDefinition)
```
Comment (tzolov) [inline spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/support/ArgumentSanitizer.java]: To add ArgumentSanitizerTest test for `sanitizeTask`

**Comment by tzolov** [comment]
@ilayaperumalg i've added the missing [TaskDefinitionToDslConverterTests](https://github.com/tzolov/spring-cloud-dataflow/commit/2885992934c5b3391d9a29e57343945b0e576175) which reviled that `TaskDefinitionToDslConverter` can not reverse engineer Composite Tasks!

Furthermore because the TaskDefinition design doesn't provide first-class support/representation for composite tasks graphs, the reverse engineering of such tasks could be a very daunting and error prone job.

**Comment by ilayaperumalg** [comment]
@tzolov sure, please feel free add any test cases on the dsl converter tests. I added some in this PR.

**Comment by tzolov** [comment]
Thanks @ilayaperumalg. I will add a non composed task Assert to the Dsl methods and will merge the result.

**Comment by tzolov** [comment]
Added check to fail on CRT, rebased and merged at 120d10b55cb59a3f87e09a3871c61ebc189598ed

**Pull request closed without merging**